File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/NotSubmittableNotification.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.services.PluginSettings;

public class NotSubmittableNotification extends Notification {

  /**
   * Construct a notification that explains that the exercise cannot be submitted from the plugin.
   */
  public NotSubmittableNotification() {
    super(
        PluginSettings.A_PLUS,
        getText("notification.NotSubmittableNotification.title"),
        getText("notification.NotSubmittableNotification.content"),
        NotificationType.ERROR
    );
  }

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/NotSubmittableNotification.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

/**
 * A notification that explains that the exercise cannot be submitted from the plugin.
 */
class NotSubmittableNotification : Notification(
    PluginSettings.A_PLUS,
    MyBundle.message("notification.NotSubmittableNotification.title"),
    MyBundle.message("notification.NotSubmittableNotification.content"),
    NotificationType.ERROR
)
